### PR TITLE
Docs: Replace usage of yarn in readmes

### LINKED ISSUE #237: Chore: Update package manager in readme for the examples
### Descriptions

Many of our examples has documentation that reference to `yarn` as the package manger even if we have changed to `npm` as the default package manager.

### Todo

- [ ] [App with backend](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-backend)
- [ ] [App with dashboards](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-dashboards)
- [ ] [Datasource basic](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-basic)
- [ ] [Datasource http backend](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-http-backend)
- [ ] [Datasource streaming backend websocket](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-streaming-backend-websocket) 
- [ ] [Datasource streaming websocket](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-streaming-websocket)
- [ ] [Panel basic](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/panel-basic)
- [ ] [Panel datalinks](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/panel-datalinks)
- [ ] [Panel plotly](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/panel-plotly)

### PR DESCRIPTION
**Pull request by jackw**
This PR replaces all usage of `yarn` in favour of `npm` across all readmes. It also adds an admonition to the root readme explaining the examples use NPM and that we don't offer to support using them with alternative package managers.

fixes: #237

### CODE REVIEW

**Review by academo — APPROVED** [review]
Great work. 

How do you feel about updating this script too?

https://github.com/grafana/grafana-plugin-examples/blob/main/scripts/update-grafana-version.sh

**Review by leventebalogh — APPROVED** [review]
Nice one 👍

File: README.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ This repository contains example plugins to showcase different use cases.
 | [datasource-streaming-backend-websocket](examples/datasource-streaming-backend-websocket) | Shows how to create an event-based data source plugin using backend streams.                             |
 | [datasource-basic](examples/datasource-basic)                             | Shows how to build a basic data source plugin.                                                                |
 
+> [!NOTE]
+> The plugin examples in this repository use NPM to manage frontend dependencies. Whilst you are welcome to copy these examples and use Yarn or PNPM instead we offer no support for them.
```
Comment (xnyo) [inline README.md:39]: Nit:

```suggestion
> The plugin examples in this repository use NPM to manage frontend dependencies. Whilst you are welcome to copy these examples and use Yarn or PNPM instead, we offer no support for them.
```

**Review by xnyo — APPROVED** [review]
Looks good! :smile: 

With a follow-up PR we can also remove this code block in `update-grafana-version.sh`:

https://github.com/grafana/grafana-plugin-examples/blob/b263aee20dd8470e8de839c3a862ed7fe9c6231d/scripts/update-grafana-version.sh#L65-L73

**Comment by jackw** [comment]
> Looks good! 😄
> 
> With a follow-up PR we can also remove this code block in `update-grafana-version.sh`:
> 
> https://github.com/grafana/grafana-plugin-examples/blob/b263aee20dd8470e8de839c3a862ed7fe9c6231d/scripts/update-grafana-version.sh#L65-L73

@xnyo I think I covered that in https://github.com/grafana/grafana-plugin-examples/pull/243/commits/2365dc190937e7ceb3634ec4303680d3d4c5e323? Let me know if I missed something.

**Pull request merged**